cli/src/app.rs

```diff
@@ -210,7 +211,7 @@ pub fn app<'a>() -> clap::Command<'a> {
                     .long("lockfile-type")
                     .value_name("type")
                     .help("The type of the lock file (default: auto)")
-                    .possible_values(parse::lockfile_types()),
+                    .value_parser(LockfileValueParser),
```

Comment:
Is there a reason to implement our own `ValueParser` rather than simply using [`PossibleValuesParser`](https://docs.rs/clap/4.0.9/clap/builder/struct.PossibleValuesParser.html)?

Reply:
Oh wow, so if you pass a `Vec` to `value_parser` it doesn't work, but if you wrap it in `PossibleValuesParser` it does. So `value_parser` only accepts owned slices.